### PR TITLE
First steps to allowing Linux to use EGL and OpenGLES

### PR DESCRIPTION
**Pull request by prashantgupta24**
[Review on Reviewable](https://reviewable.io/reviews/servo/rust-layers/220)

### CODE REVIEW

**Comment by glennw** [comment]
Reviewed 2 of 4 files at r1, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/platform/egl/surface.rs, line 71 \[r4\]](https://reviewable.io:443/reviews/servo/rust-layers/220#-K4wUVeVkgvabsLPw5HV:-K4wUVeVkgvabsLPw5HW:-1878314577) ([raw file](https://github.com/servo/rust-layers/blob/cd78d85621b15a5e6056e3dd3504b82d074ed741/src/platform/egl/surface.rs#L71)):*
BGRA constant only exists on Linux - we'll need to apply the same trick I have done here (https://github.com/glennw/webrender/blob/master/src/device.rs#L16) to support BGRA on Linux and BGRA_EXT on Android.

---

*[src/platform/egl/surface.rs, line 85 \[r4\]](https://reviewable.io:443/reviews/servo/rust-layers/220#-K4wUdZuN1bZ2A98EDrh:-K4wUdZuN1bZ2A98EDri:994361279) ([raw file](https://github.com/servo/rust-layers/blob/cd78d85621b15a5e6056e3dd3504b82d074ed741/src/platform/egl/surface.rs#L85)):*
Tabs -> spaces

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/rust-layers/220)*

**Comment by glennw** [comment]
Just a few minor changes left - nearly there! :)

**Comment by glennw** [comment]
This looks good to me (well, it will still need a rebase + squash) - but I'd like to get @mrobinson to check over it too.

File: src/platform/linux/surface.rs [hunk 1]
```diff
@@ -141,11 +160,22 @@ impl NativeDisplay {
     }
 
     pub fn platform_display_data(&self) -> PlatformDisplayData {
-        PlatformDisplayData {
-            display: self.display,
-            visual_info: self.visual_info,
+        match *self {
+            NativeDisplay::Glx(info) => {
```
Comment (mrobinson) [inline src/platform/linux/surface.rs]: This should still be called GLX instead of Glx.

File: src/platform/linux/surface.rs [hunk 2]
```diff
@@ -141,11 +160,22 @@ impl NativeDisplay {
...
+                PlatformDisplayData {
+                    display: info.display,
+                    visual_info: info.visual_info,
+                }
+            }
+            NativeDisplay::Egl(_) => unreachable!(),
```
Comment (mrobinson) [inline src/platform/linux/surface.rs]: And here EGL instead of Egl.

File: src/platform/linux/surface.rs [hunk 3]
```diff
@@ -141,11 +160,22 @@ impl NativeDisplay {
...
         }
     }
+
+    pub fn from_es2() -> NativeDisplay {
```
Comment (mrobinson) [inline src/platform/linux/surface.rs]: I think maybe this should be called new_egl_display instead of from_es2. GLES2 is a different concept than EGL.

**Comment by mrobinson** [comment]
Looks good to me, apart from a few minor style and naming issues.

**Comment by jdm** [comment]
Rebased, squashed and comments addressed in #232.

**Pull request closed without merging**